python/lsst/ip/isr/overscan.py

```diff
@@ -0,0 +1,504 @@
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import numpy as np
+import lsst.afw.math as afwMath
+import lsst.afw.image as afwImage
+import lsst.pipe.base as pipeBase
+import lsst.pex.config as pexConfig
+import lsst.pex.exceptions as pexExcept
+
+__all__ = ["OverscanCorrectionTaskConfig", "OverscanCorrectionTask"]
+
+
+class OverscanCorrectionTaskConfig(pexConfig.Config):
+    """Overscan correction options.
+    """
+    fitType = pexConfig.ChoiceField(
+        dtype=str,
+        doc="The method for fitting the overscan bias level.",
+        default='MEDIAN',
+        allowed={
+            "POLY": "Fit ordinary polynomial to the longest axis of the overscan region",
+            "CHEB": "Fit Chebyshev polynomial to the longest axis of the overscan region",
+            "LEG": "Fit Legendre polynomial to the longest axis of the overscan region",
+            "NATURAL_SPLINE": "Fit natural spline to the longest axis of the overscan region",
+            "CUBIC_SPLINE": "Fit cubic spline to the longest axis of the overscan region",
+            "AKIMA_SPLINE": "Fit Akima spline to the longest axis of the overscan region",
+            "MEAN": "Correct using the mean of the overscan region",
+            "MEANCLIP": "Correct using a clipped mean of the overscan region",
+            "MEDIAN": "Correct using the median of the overscan region",
+            "MEDIAN_PER_ROW": "Correct using the median per row of the overscan region",
+        },
+    )
+    order = pexConfig.Field(
+        dtype=int,
+        doc=("Order of polynomial or to fit if overscan fit type is a polynomial, " +
+             "or number of spline knots if overscan fit type is a spline."),
+        default=1,
+    )
+    numSigmaClip = pexConfig.Field(
+        dtype=float,
+        doc="Rejection threshold (sigma) for collapsing overscan before fit",
+        default=3.0,
+    )
+    overscanIsInt = pexConfig.Field(
+        dtype=bool,
+        doc="Treat overscan as an integer image for purposes of overscan.FitType=MEDIAN" +
+            " and overscan.FitType=MEDIAN_PER_ROW.",
+        default=True,
+    )
+
+
+class OverscanCorrectionTask(pipeBase.Task):
+    """Correction task for overscan.
+
+    This class contains a number of utilities that are easier to
+    understand and use when they are not embedded in nested if/else
+    loops.
+
+    Parameters
+    ----------
+    statControl : `lsst.afw.math.StatisticsControl`, optional
+        Statistics control object.
+    """
+    ConfigName = OverscanCorrectionTaskConfig
+    _DefaultName = "overscan"
+
+    def __init__(self, statControl=None, **kwargs):
+        super().__init__(**kwargs)
+        if statControl:
+            self.statControl = statControl
+        else:
+            self.statControl = afwMath.StatisticsControl()
+
+    def run(self, ampImage, overscanImage):
+        self.statControl.setNumSigmaClip(self.config.numSigmaClip)
```

Comment:
I don't think you should be modifying something that's been given to you. Your API seems to want things both ways, but you should choose one: use a `statControl` that you're given, or configure a new `statControl` from the configuration.